compiler/vm/vmops.nim

```diff
@@ -223,19 +222,11 @@ proc getEffectList(cache: IdentCache, idgen: IdGenerator; a: VmArgs;
 template writeResult(ret) {.dirty.} =
   writeTo(ret, a.getResultHandle(), a.mem[])
 
-# XXX: various callbacks currently require captured state (i.e. closures)
-#      in order to work. Once `TCtx` is split up into smaller parts, the VM
-#      environment related ones could be passed to the callbacks instead,
-#      making most of the capturing unnecessary
-
 proc registerBasicOps*(c: var TCtx) =
   ## Basic system operations as well as callbacks for some stdlib functions
   ## that don't interact with the host environement, but use language features
   ## that the VM doesn't directly support (such as 'importc'-ed functions)
 
-  # captured vars:
-  let isJs = c.config.backend == backendJs
```

Comment:
With the removal of captures should the calling convention be explicit in order to avoid future captures?

Reply:
Ah, good call, I'll adjust the definitions.

Reply:
I've changed all callback procedures to use the `.nimcall` calling convention.